Sources/Plasma/PubUtilLib/plVault/plVaultClientApi.cpp

```diff
@@ -367,57 +324,51 @@ static void BuildNodeTree (
 ) {
     for (unsigned i = 0; i < refCount; ++i) {
         // Find/Create global links
-        RelVaultNodeLink * parentLink = s_nodes.Find(refs[i].parentId);
-        if (!parentLink) {
+        auto parentIt = s_nodes.find(refs[i].parentId);
+        if (parentIt == s_nodes.end()) {
             newNodeIds->emplace_back(refs[i].parentId);
-            parentLink = new RelVaultNodeLink(false, 0, refs[i].parentId);
-            parentLink->node->SetNodeId_NoDirty(refs[i].parentId);
-            s_nodes.Add(parentLink);
-        }
-        else {
+            hsRef<RelVaultNode> newParentNode(new RelVaultNode(), hsStealRef);
+            newParentNode->SetNodeId_NoDirty(refs[i].parentId);
+            parentIt = s_nodes.emplace(refs[i].parentId, std::move(newParentNode)).first;
```

Comment:
Shouldn't `parentIt` be set to the iterator returned by `emplace()`? It looks like `parentIt` is currently an iterator for `s_nodes` :/

Reply:
That's what the code does. Not sure I understand the question :/

The `.first` at the end is because [`std::unordered_map::emplace`](https://en.cppreference.com/w/cpp/container/unordered_map/emplace) actually returns a `std::pair<iterator, bool>`.

Reply:
Ah, right, I was thinking `s_nodes` was an `std::list` for some reason.